Add script to build and run shake script

tuncer opened the pull request:
This is based on Neil Mitchell's demo project docs.

moosotc commented:
I am not sure why this is needed, simple:
$ runghc Build.hs [options]
does the trick for me (and builds by default in the directory named "build" (automatically 
created by Shake))

tuncer commented:
It's a reasonable suggestion from Neil's docs, and it hides some details one might get wrong when calling manually. It will rebuild the executable if you modify Build.hs and pass through the args, without requiring `ghc ... && ./Build _ARGS_`. I've rebased and simplified the script.
